dals/boost/azureStorage.go

```
@@ -0,0 +1,68 @@
+package boost
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/Azure/azure-storage-queue-go/azqueue"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/searchspring/nebo/common"
+	"log"
+	"net/url"
+	"time"
+)
+
+// Golang Azure Queue docs
+// https://pkg.go.dev/github.com/Azure/azure-storage-queue-go/azqueue?utm_source=godoc#pkg-overview
+
+type ICloudQueue interface {
+	EnqueueMessage(queue string, cloudEvent CloudEvent) (string, error)
+}
+
+type AzureStorage struct {
+	AccountName      string
+	ConnectionString string
+}
+
+// NewAzureStorage AzureStorage constructor that references ENV variables
+func NewAzureStorage() AzureStorage {
+	var env common.EnvVars
+	envconfig.Process("", &env)
+
+	return AzureStorage{
+		AccountName:      env.AzureAccount,
+		ConnectionString: env.AzureConnection,
+	}
+}
+
+func (storage *AzureStorage) EnqueueMessage(queue string, cloudEvent CloudEvent) (string, error) {
+	credentials, err := azqueue.NewSharedKeyCredential(storage.AccountName, storage.ConnectionString)
+
+	if err != nil {
+		log.Println("Failed to authenticate")
+		return fmt.Sprintf("Failed to authenticate to Azure resource: `%v`", storage.AccountName), err
+	}
+
+	pipeline := azqueue.NewPipeline(credentials, azqueue.PipelineOptions{})
+
+	u, _ := url.Parse(fmt.Sprintf("https://%s.queue.core.windows.net", storage.AccountName))
+	serviceURL := azqueue.NewServiceURL(*u, pipeline)
+
+	mainDispatchQueue := serviceURL.NewQueueURL(queue)
+	ctx := context.TODO() // This example uses a never-expiring context.
+
+	messagesURL := mainDispatchQueue.NewMessagesURL()
+
+	bytes, _ := json.Marshal(cloudEvent)
```

Best to not ignore errors

---

most methods seem to return `(someType, error)` is there a cleaner way to handle errors without every other line in your code being if `err != nil`

---

Not really. This is a go-ism. Not too different from other languages having if x != null, etc.

Personally, I don't mind the error as a type (compared to null/nil checks), but I hate that the language doesn't have something more advanced to reduce the noise of the if err != nil. Something like computation expressions (monads) in F# would be super valuable to just let error "fall out" automagically. But alas, Google does not decree it so